### PR TITLE
move typing to common module to prepare for the prototype of spdx 3.0

### PR DESCRIPTION
**Pull request by meretp**
Signed-off-by: Meret Behrens <meret.behrens@tngtech.com>

### CODE REVIEW

File: src/spdx/parser/jsonlikedict/dict_parsing_functions.py [hunk 1]
```diff
@@ -12,7 +12,7 @@
 
 from spdx.model.spdx_no_assertion import SpdxNoAssertion
 from spdx.model.spdx_none import SpdxNone
-from spdx.model.typing.constructor_type_errors import ConstructorTypeErrors
+from common.typing.type_checks import ConstructorTypeErrors
```
Comment (armintaenzertng) [inline src/spdx/parser/jsonlikedict/dict_parsing_functions.py:15]: Why are `ConstructorTypeErrors` imported from `type_checks` now? Does this even work?

File: src/spdx/parser/jsonlikedict/relationship_parser.py [hunk 2]
```diff
@@ -11,7 +11,7 @@
 from typing import Dict, List, Optional
 
 from spdx.model.relationship import Relationship, RelationshipType
-from spdx.model.typing.constructor_type_errors import ConstructorTypeErrors
+from common.typing.type_checks import ConstructorTypeErrors
```
Comment (armintaenzertng) [inline src/spdx/parser/jsonlikedict/relationship_parser.py:14]: Same as above

File: src/spdx/parser/jsonlikedict/dict_parsing_functions.py (hunk 1 above)
Comment (meretp) [reply src/spdx/parser/jsonlikedict/dict_parsing_functions.py]: Good catch! Fixed this.

**Review by armintaenzertng — APPROVED** [review]
Looks good to me now :)

**Pull request merged**